### PR TITLE
feat: use owner reference with policy server resources.

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Updates the controller to set owner references for all the resources created to make the policy server works. Which includes: service, configmap, deployment and secret. Therefore, we can remove the reconciliation loop code to remove this resources when the user deletes the policy server and rely on Kubernetes garbage collector to do the job.

Fix https://github.com/kubewarden/kubewarden-controller/issues/660

## Test

```shell
make test
# manual testing with tilt
tilt up --stream
# install and uninstall policy server
```

### CODE REVIEW

File: controllers/policyserver_controller_test.go [hunk 1]
```diff
@@ -137,16 +226,47 @@ var _ = Describe("PolicyServer controller", func() {
...
+				policyServer, err := getTestPolicyServer(policyServerName)
+				Expect(err).Should(Succeed())
+				controllerutil.RemoveFinalizer(policyServer, IntegrationTestsFinalizer)
+				err = reconciler.Client.Update(ctx, policyServer)
+				Expect(err).ToNot(HaveOccurred())
+
+				Eventually(func() error {
+					_, err := getTestPolicyServer(policyServerName)
+					return err
+				}, timeout, pollInterval).ShouldNot(Succeed())
```
Comment (flavio) [inline controllers/policyserver_controller_test.go:249]: Can you refactor that to make sure it fails with the error "not found"?

**Comment by jvanz** [comment]
This PR is based on top of #730. Waiting for that PR to get merged before moving this to pending review.

**Comment by jvanz** [comment]
@kubewarden/kubewarden-developers the integration tests are failing but I do not think this is related to the changes here. It looks like something different. I'm moving this PR to be reviewed by the team while I investigate the issue. It may take some time because I cannot reproduce this in my local branch. :crossed_fingers:

File: internal/pkg/admission/policy-server_integration_test.go [hunk 2]
```diff
@@ -1,95 +0,0 @@
...
-	"testing"
-	"time"
-
-	"github.com/kubewarden/kubewarden-controller/internal/pkg/admissionregistration"
-	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
-)
-
-const port = "8181"
-
-func TestCAAndCertificateCreationInAHttpsServer(t *testing.T) {
```
Comment (viccuad) [inline internal/pkg/admission/policy-server_integration_test.go:18]: Should we drop this test?

**Review by viccuad — COMMENTED** [review]
LGTM, just one question about the cert test refactor.

File: internal/pkg/admission/policy-server_integration_test.go (hunk 2 above)
Comment (jvanz) [reply internal/pkg/admission/policy-server_integration_test.go:18]: Fair enough, I've re added this test in the internal/pkg/admission/policy-server-ca-secret_test.go file. ;)

File: config/crd/bases/policies.kubewarden.io_admissionpolicies.yaml [hunk 3]
```diff
@@ -163,11 +163,13 @@ spec:
                           items:
                             type: string
                           type: array
+                          x-kubernetes-list-type: atomic
```
Comment (flavio) [inline config/crd/bases/policies.kubewarden.io_admissionpolicies.yaml:166]: Where is this coming from?

**Review by flavio — APPROVED** [review]
LGTM, I left one minor comment

File: config/crd/bases/policies.kubewarden.io_admissionpolicies.yaml (hunk 3 above)
Comment (jvanz) [reply config/crd/bases/policies.kubewarden.io_admissionpolicies.yaml:166]: This came from a recent [update](https://github.com/kubernetes/apimachinery/commit/bfbbdecc1d6d161ce84da90c4a58e21c6eb8261f) in the apimachinery package. As we use label selector objects in the policy spec, the CRDs will be updated with the latest changes there as well. I've added a commit here to fix that in our main branch and avoid generating noise every time that we run the controller locally.  

If you think is better, I can open another PR just for that update in the CRDs.

**Pull request merged**